scan/lib/scan/test_command_generator.rb

```diff
@@ -49,7 +49,9 @@ def actions
         actions = []
         actions << :clean if config[:clean]
         actions << :build unless config[:skip_build]
-        actions << :test
+        actions << "build-for-testing" if config[:build_for_testing]
+        actions << :test unless config[:test_without_building] || config[:build_for_testing]
+        actions << "test-without-building" if config[:test_without_building]
```

Comment:
Okay, there is not many alternatives. Maybe just splitting it into more lines makes it easier 👍 
```
if config[:build_for_testing]
  actions << "build-for-testing"
end

unless config[:test_without_building] || config[:build_for_testing])
  actions << :test
end

if config[:test_without_building]
  actions << "test-without-building"
end
```